core/queryalgebra/evaluation/src/main/java/org/eclipse/rdf4j/query/algebra/evaluation/ArrayBindingSet.java

```diff
@@ -214,6 +214,12 @@ public Set<String> getBindingNames() {
 
 	@Override
 	public Value getValue(String bindingName) {
+		for (int i = 0; i < bindingNames.length; i++) {
+			if (bindingNames[i] == bindingName && whichBindingsHaveBeenSet[i]) {
+				return values[i];
+			}
+		}
+
 		for (int i = 0; i < bindingNames.length; i++) {
 			if (bindingNames[i].equals(bindingName) && whichBindingsHaveBeenSet[i]) {
 				return values[i];
```

Comment:
There is a high chance that the bindingName string passed into this method is the same string object as in our bindingNames array. Looping through an array or strings and checking the elements with `==` is much faster than `.equals(...)`. If we don't find a match on our first pass we have a second for-loop with `.equals(...)` as a fallback.

See: https://github.com/eclipse/rdf4j/pull/3943/files#r887778841